mono/metadata/reflection.c

```diff
@@ -13516,6 +13517,9 @@ reflection_initialize_generic_parameter (MonoReflectionGenericParam *gparam, Mon
 
 	mono_error_init (error);
 
+	// Further down we check whether gparam->tbuilder != NULL.
```

Comment:
In addition, the compiler is permitted to optimise out that check because of this access.